src/util/generateBatchPlateData.ts

```diff
@@ -0,0 +1,222 @@
+    record.systemNumber = `BPS${paddedIterator}`;
+    record.createdTimestamp = date;
+    record.vin = `BPV${paddedIterator}`;
+    record.partialVin = paddedIterator;
+    record.primaryVrm = `${paddedIterator}Z`;
+
+    records.push(record);
+    triggerData.push({ systemNumber: record.systemNumber, createdtimestamp: record.createdTimestamp });
+
+    if (i % 1000 === 0) {
+      const fileName = `/tmp/technical-records-v3-with-batch-plates-${fileCount.toString()}.json`;
+      fileNames.push(fileName);
+      fs.writeFileSync(fileName, JSON.stringify(records, null, 2));
+      records = [];
+      fileCount++;
+    }
+  }
+
+  records = [];
+  // Missing functionCode
+  for (let i = 80001; i <= 80500; i++) {
+    const record = JSON.parse(JSON.stringify(completeTRLTechRecords[0])) as TechRecordGETTRL;
+    const paddedIterator = padZeroes(i.toString(), 6);
+    record.systemNumber = `BPS${paddedIterator}`;
+    record.createdTimestamp = date;
+    record.vin = `BPV${paddedIterator}`;
+    record.partialVin = paddedIterator;
+    delete record.techRecord_functionCode;
+    records.push(record);
+    triggerData.push({ systemnumber: record.systemNumber, createdtimestamp: record.createdTimestamp });
+  }
+
+  const fileName1 = `/tmp/technical-records-v3-with-batch-plates-${fileCount.toString()}.json`;
+  fileNames.push(fileName1);
+  fs.writeFileSync(fileName1, JSON.stringify(records, null, 2));
+  records = [];
+  fileCount++;
+
+  // missing model and functionCode
+  for (let i = 80501; i <= 90000; i++) {
+    const record = JSON.parse(JSON.stringify(completeTRLTechRecords[0])) as TechRecordGETTRL;
+    const paddedIterator = padZeroes(i.toString(), 6);
+    record.systemNumber = `BPS${paddedIterator}`;
+    record.createdTimestamp = date;
+    record.vin = `BPV${paddedIterator}`;
+    record.partialVin = paddedIterator;
+    delete record.techRecord_functionCode;
+    delete record.techRecord_model;
+    records.push(record);
+    triggerData.push({ systemnumber: record.systemNumber, createdtimestamp: record.createdTimestamp });
+
+    if (i % 1000 === 0) {
+      const fileName = `/tmp/technical-records-v3-with-batch-plates-${fileCount.toString()}.json`;
+      fileNames.push(fileName);
+      fs.writeFileSync(fileName, JSON.stringify(records, null, 2));
+      records = [];
+      fileCount++;
+    }
+  }
+
+  records = [];
+
+  // missing variantNumber
+  for (let i = 90001; i <= 98000; i++) {
+    const record = JSON.parse(JSON.stringify(completeHGVTechRecord)) as TechRecordGETHGV;
+    const paddedIterator = padZeroes(i.toString(), 6);
+    record.systemNumber = `BPS${paddedIterator}`;
+    record.createdTimestamp = date;
+    record.vin = `BPV${paddedIterator}`;
+    record.partialVin = paddedIterator;
+    record.primaryVrm = `${paddedIterator}Z`;
+    delete record.techRecord_variantNumber;
+
+    records.push(record);
+    triggerData.push({ systemnumber: record.systemNumber, createdtimestamp: record.createdTimestamp });
```

Comment:
In regards to my comment on trigger-data.json, just wondering if there is a reason this is not systemNumber and createdTimestamp?

Reply:
caught up with Leo, think it's a boo boo, good spot